controllers/idle_handlers.go

```diff
@@ -18,15 +18,19 @@ package controllers
 
 import (
 	"context"
+	"fmt"
 
 	ranv1alpha1 "github.com/openshift-kni/lifecycle-agent/api/v1alpha1"
+	"github.com/openshift-kni/lifecycle-agent/controllers/utils"
 	ctrl "sigs.k8s.io/controller-runtime"
 )
 
 func (r *ImageBasedUpgradeReconciler) handleAbort(ctx context.Context, ibu *ranv1alpha1.ImageBasedUpgrade) (ctrl.Result, error) {
-
-	// TODO actual steps
-	// If succeeds, return doNotRequeue
+	ostreeUndeployCommand := fmt.Sprintf("ostree admin undeploy rhcos_%s", ibu.Spec.SeedImageRef.Version)
```

Comment:
The undeploy command takes the deployment index, not the stateroot

Reply:
I have tested this on 4.10.15 and it works with the name. Maybe it has changed in newer versions.

Reply:
I think it is an undocumented feature
`ostree admin undeploy rhcos_4.10.15
Bootloader updated; bootconfig swap: yes; deployment count change: -1
Deleted deployment 3f573c36a90b1c2412487be2401c2c0442c75da829c26182a59029494279739c.0
`

Reply:
I think we'd should confirm with @cgwalters or someone else from the coreos team if we want to use an "undocumented feature"

Reply:
I think we should also be confirming the existence of the stateroot and deployment first to determine whether there's any cleanup required.

Reply:
@javipolo is the ostree plugin going to provide this function? will it be based on index or name?

Reply:
@sudomakeinstall2 What Don mentioned in his last comment is essentially what I was thinking for CNF-10134. This kind of data seems to be generally useful for multiple stage handlers. Therefore it makes sense to implement it as a part of main reconcile logic before entering stage specific parts.

Reply:
> @javipolo is the ostree plugin going to provide this function? will it be based on index or name?

what do you mean by the ostree plugin?
I think we'll need someone from RHCOS to clarify ostree functionality here ... maybe @jmarrero can provide some insights?

Reply:
there is `ibu-imager/internal/ostreeclient` but it is internal to IBU.

Reply:
> > @javipolo is the ostree plugin going to provide this function? will it be based on index or name?
> 
> what do you mean by the ostree plugin? I think we'll need someone from RHCOS to clarify ostree functionality here ... maybe @jmarrero can provide some insights?

@javipolo : @jc-rh is referring to the plugin for setting up and managing the new stateroot and deployment in CNF-9797

Reply:
updated

Reply:
I think you just stumbled on part of the code that needs some input sanitization.
I don't see any test where we are testing passing anything other than a number and giving the code a glance [here](https://github.com/ostreedev/ostree/blob/main/src/ostree/ot-admin-builtin-undeploy.c#L53C28-L53C44), I think we are just taking whatever the input it and trying to convert it to int.

Trying really quick to replicate:
```
jmarrero@fedora ~> sudo ostree admin undeploy trash
[sudo] password for jmarrero: 
error: Cannot undeploy currently booted deployment 0
```

I bet that deployment that is being undeployed index is 0?

Reply:
yes I did realize that after I looked at the code, and changed the implementation.